lib/datadog/ci/configuration/components.rb

```diff
@@ -118,6 +110,42 @@ def activate_ci!(settings)
           )
         end
 
+        def build_test_optimisation(settings, test_visibility_api)
+          if settings.ci.itr_code_coverage_use_single_threaded_mode &&
+              settings.ci.itr_test_impact_analysis_use_allocation_tracing
+            Datadog.logger.warn(
+              "Intelligent test runner: Single threaded coverage mode is incompatible with allocation tracing. " \
+              "Allocation tracing will be disabled. It means that test impact analysis will not be able to detect " \
+              "instantiations of objects in your code, which is important for ActiveRecord models. " \
+              "Please add your app/model folder to the list of tracked files or disable single threaded coverage mode."
+            )
```

Comment:
If you ever want to support this -- you can use `rb_thread_current()` inside the tracepoint to compare it to 'th_covered` and ignore any events from threads that aren't your target.

Reply:
I will hold off this for now as I am not sure if it's allowed inside the tracepoint ;)

Reply:
Indeed, up to you! (I do suspect it's safe; as an alternative, you could always do some magic with thread-locals, e.g. by identifying which native thread matches the Ruby thread and then safely check in which native thread you landed during the hook)